HARK/ConsumptionSaving/ConsRiskyContribModel.py

```diff
@@ -0,0 +1,1768 @@
+    vFunc_Reb_Fxd : ValueFunc3D
+        Stage value function over normalized liquid resources, normalized
+        iliquid resources, and income contribution share when the agent is
+        not able to adjust his portfolio.
+    dfracFunc_Fxd : Interp2D
+        Deposit function over normalized liquid resources, normalized iliquid
+        resources, and income contribution share when the agent is not able to
+        adjust his portfolio.
+        Must be ConstantFunction(0.0)
+    dvdmFunc_Reb_Fxd : MargValueFunc3D
+        Marginal value over normalized liquid resources when the agent is not
+        able to adjust his portfolio.
+    dvdnFunc_Reb_Fxd : MargValueFunc3D
+        Marginal value over normalized iliquid resources when the agent is not
+        able to adjust his portfolio.
+    dvdsFunc_Reb_Fxd : Interp3D
+        Marginal value function over income contribution share when the agent
+        is not able to ajust his portfolio.
+    """
+
+    distance_criteria = ["dvdmFunc_Reb_Adj", "dvdnFunc_Reb_Adj"]
+
+    def __init__(
+        self,
+        # Rebalancing stage, adjusting
+        vFunc_Reb_Adj=None,
+        dfracFunc_Adj=None,
+        dvdmFunc_Reb_Adj=None,
+        dvdnFunc_Reb_Adj=None,
+        # Rebalancing stage, fixed
+        vFunc_Reb_Fxd=None,
+        dfracFunc_Fxd=None,
+        dvdmFunc_Reb_Fxd=None,
+        dvdnFunc_Reb_Fxd=None,
+        dvdsFunc_Reb_Fxd=None,
+    ):
+
+        # Rebalancing stage
+        if vFunc_Reb_Adj is None:
+            vFunc_Reb_Adj = NullFunc()
+        if dfracFunc_Adj is None:
+            dfracFunc_Adj = NullFunc()
+        if dvdmFunc_Reb_Adj is None:
+            dvdmFunc_Reb_Adj = NullFunc()
+        if dvdnFunc_Reb_Adj is None:
+            dvdnFunc_Reb_Adj = NullFunc()
+
+        if vFunc_Reb_Fxd is None:
+            vFunc_Reb_Fxd = NullFunc()
+        if dfracFunc_Fxd is None:
+            dfracFunc_Fxd = NullFunc()
+        if dvdmFunc_Reb_Fxd is None:
+            dvdmFunc_Reb_Fxd = NullFunc()
+        if dvdnFunc_Reb_Fxd is None:
+            dvdnFunc_Reb_Fxd = NullFunc()
+        if dvdsFunc_Reb_Fxd is None:
+            dvdsFunc_Reb_Fxd = NullFunc()
+
+        # Components of the adjusting problem
+        self.vFunc_Reb_Adj = vFunc_Reb_Adj
+        self.dfracFunc_Adj = dfracFunc_Adj
+        self.dvdmFunc_Reb_Adj = dvdmFunc_Reb_Adj
+        self.dvdnFunc_Reb_Adj = dvdnFunc_Reb_Adj
+
+        # Components of the fixed problem
+        self.vFunc_Reb_Fxd = vFunc_Reb_Fxd
+        self.dfracFunc_Fxd = dfracFunc_Fxd
+        self.dvdmFunc_Reb_Fxd = dvdmFunc_Reb_Fxd
+        self.dvdnFunc_Reb_Fxd = dvdnFunc_Reb_Fxd
+        self.dvdsFunc_Reb_Fxd = dvdsFunc_Reb_Fxd
+
+
+# Class for the contribution share stage solution
+class RiskyContribShaSolution(MetricObject):
+    """
+    A class for representing the solution to the contribution-share stage of
+    the 'RiskyContrib' model.
+    
+    Parameters
+    ----------
+    vFunc_Sha_Adj : ValueFunc2D
```

Comment:
similarly... I think it's better to reduce the number of redundant `Sha` tags

Reply:
Agreed. It's residual notation from a previous iteration where there weren't split classes. Addressed in https://github.com/econ-ark/HARK/pull/832/commits/6e6f961d033e2cdd8474bb90b969debdd98a31c1

Thank you!